wagtail/coreutils.py

```diff
@@ -422,11 +422,12 @@ def get_dummy_request(*, path: str = "/", site: "Site" = None) -> HttpRequest:
     if site:
         server_name = site.hostname
         server_port = site.port
-    elif settings.ALLOWED_HOSTS == ["*"]:
-        server_name = "example.com"
     else:
         server_name = settings.ALLOWED_HOSTS[0]
 
+        if server_name == "*":
```

Comment:
This wouldn't work if the `"*"` is not the first item in the list

Reply:
That's fine, isn't it? If the first item in the list is a real hostname (and not `"*"`), we want to use that instead of `example.com`.

Reply:
Oh, yep, you're right. I misunderstood what the function is doing with it. Thanks!